### PR TITLE
Pdf session connection

### PR DESCRIPTION
**Pull request by sarinali**
# Description

Created endpoint for linking PDF uploads to the current active session

## Type of change

- [x] New feature (non-breaking change which adds functionality)

https://github.com/Callhub-Connect/callhub-backend/assets/74844953/7bf20a0c-f5e2-4459-924e-7a636312fb1b

# How Has This Been Tested?

Tested using two JUnit Tests:

```java
@Test
    void testPDFLinkValidSession() throws Exception {
        String currentDirectory = System.getProperty("user.dir");
        String pathName = currentDirectory + "/src/main/java/callhub/connect/pdfs/fall2023.pdf";
        File pdfFile = new File(pathName);
        MockMultipartFile sampleFile = new MockMultipartFile(
                "file",
                pdfFile.getName(),
                "application/pdf",
                new FileInputStream(pdfFile));


        when(documentRepository.save(any())).thenReturn(new FileDocument(
                "MockFile",
                new Binary("sample pdf content".getBytes()),
                LocalDate.now()));

        Session mockSession = new Session(true, "ABCDEF");

        // Mock the behavior of the sessionRepository.save() method
        when(sessionRepository.getSessionsByActiveAndCode(true, "ABCDEF")).thenReturn(mockSession);

        MockMultipartHttpServletRequestBuilder multipartRequest =
                MockMvcRequestBuilders.multipart("/files/session_add_pdf");
        mockMvc.perform(multipartRequest.file(sampleFile).param("session", "ABCDEF").param("name", "MockFile")).andExpect(status().isOk());
    }

    @Test
    void testPDFLinkInvalidSession() throws Exception {
        String currentDirectory = System.getProperty("user.dir");
        String pathName = currentDirectory + "/src/main/java/callhub/connect/pdfs/fall2023.pdf";
        File pdfFile = new File(pathName);
        MockMultipartFile sampleFile = new MockMultipartFile(
                "file",
                pdfFile.getName(),
                "application/pdf",
                new FileInputStream(pdfFile));


        when(documentRepository.save(any())).thenReturn(new FileDocument(
                "MockFile",
                new Binary("sample pdf content".getBytes()),
                LocalDate.now()));

        Session mockSession = new Session(true, "ABCDEF");

        // Mock the behavior of the sessionRepository.save() method
        when(sessionRepository.getSessionsByActiveAndCode(true, "ABCDEF")).thenReturn(mockSession);

        MockMultipartHttpServletRequestBuilder multipartRequest =
                MockMvcRequestBuilders.multipart("/files/session_add_pdf");
        mockMvc.perform(multipartRequest.file(sampleFile).param("session", "123456").param("name", "MockFile")).andExpect(status().isBadRequest());
    }```

### CODE REVIEW

File: src/main/java/callhub/connect/controllers/FileController.java
Comment (3milyfz) [inline src/main/java/callhub/connect/controllers/FileController.java:57]: Missing Import for LocalDate: 
- LocalDate.now() is used but there is no import statement for` java.time.LocalDate`. 
- Even if it runs fine, we should still add the import for good practice and to avoid future compilation errors!

File: src/test/java/callhub/connect/controllers/FileControllerTests.java
Comment (3milyfz) [inline src/test/java/callhub/connect/controllers/FileControllerTests.java:134]: Duplication in Test Setup: 
- There is some duplication in setting up the `MockMultipartFile` in both test methods. 
- Consider refactoring this into a separate method or a @BeforeEach setup method to reduce duplication!

File: src/test/java/callhub/connect/controllers/FileControllerTests.java
Comment (3milyfz) [inline src/test/java/callhub/connect/controllers/FileControllerTests.java:144]: Missing Import for LocalDate: 
- Similar to the other file, I would add an import statement for `java.time.LocalDate` if it's being used in the test cases.

**Review by 3milyfz — COMMENTED** [review]
Good job!
- Tested locally and works as expected
- I added a few comments on some changes I would make 
- Most are minor things including missing import statements and potential duplicate code that can be refactored

**Comment by zjayee** [comment]
Can we wait for clean architecture code to be merged first before merging this code

File: src/test/java/callhub/connect/controllers/FileControllerTests.java
Comment (zjayee) [inline src/test/java/callhub/connect/controllers/FileControllerTests.java:132]: Can we remove this file and create a blank pdf in the test setup for testing instead?

File: src/test/java/callhub/connect/controllers/FileControllerTests.java
Comment (zjayee) [inline src/test/java/callhub/connect/controllers/FileControllerTests.java:159]: Same thing here

File: src/main/java/callhub/connect/controllers/FileController.java
Comment (zjayee) [inline src/main/java/callhub/connect/controllers/FileController.java:85]: missing return

File: src/main/java/callhub/connect/controllers/FileController.java
Comment (zjayee) [inline src/main/java/callhub/connect/controllers/FileController.java:95]: missing return

**Review by zjayee — COMMENTED** [review]
Overall good with some minor comments

File: src/test/java/callhub/connect/controllers/FileControllerTests.java
Comment (sarinali) [reply src/test/java/callhub/connect/controllers/FileControllerTests.java:134]: remind me to fix this after i merge

**Review by zjayee — APPROVED** [review]
Approved + future refactoring

**Pull request merged**
